src/validation/ValidationExecutor.ts

```diff
@@ -40,6 +40,9 @@ export class ValidationExecutor {
     // -------------------------------------------------------------------------
 
     execute(object: Object, targetSchema: string, validationErrors: ValidationError[]) {
+        if (object !== Object(object)) {
```

Comment:
I don't think this is the right place to add this. I think this should add the fix to `nestedValidations()`

Reply:
@amirsaber Let's consider the next example: `validator.validate(null)`.
It will throw `TypeError: Cannot read property 'constructor' of null` without this fix.
It seems to be incorrect behavior. So we shouldn't place this check only to `nestedValidations()`.

Reply:
I agree we should handle also cases like `validator.validate(null)`. But I am a little bit worried about ending this case without any error. I think in this case we should throw an error with some message about invalid input value. What do you think @golozubov, @amirsaber